### PR TITLE
Rainbow bracket toggle button

### PR DESCRIPTION
**Pull request by TroyG-11**
https://github.com/vizhub-core/vzcode/issues/722

Working on this right now, created a button so far to toggle.

### CODE REVIEW

**Comment by TroyG-11** [comment]
No current functionality as of yet, just added the button dropdowns.

File: src/client/CodeEditor/getOrCreateEditor.ts [hunk 1]
```diff
@@ -106,6 +107,13 @@ const getAtPath = (obj, path) => {
   return current;
 };
 
+// Extend the EditorCacheValue type to include the compartments and the updateRainbowBrackets method
```
Comment (curran) [inline src/client/CodeEditor/getOrCreateEditor.ts:110]: Looks great!

**Review by curran — APPROVED** [review]
This looks like the right way to do it! Thank you so much for this.

**Review by curran — CHANGES_REQUESTED** [review]
I tested it and it seems like it does not work. Here's the screen recording of the test:


https://github.com/user-attachments/assets/c6b1a67c-a26d-4c1a-b448-c116016a3386

**Comment by TroyG-11** [comment]
> I tested it and it seems like it does not work. Here's the screen recording of the test:
> 
>  rainbowBracketsTest.mp4

Hi, I've been working on this on a separate branch right now. I had the PR up as a draft since it isn't completed yet. Will be finishing it up soon.

**Comment by TroyG-11** [comment]
Should be ready. 

It might have a quick delay before the render update on the rainbow brackets happen after switching toggle.

**Review by curran — APPROVED** [review]
LGTM! Tested locally and it works. There is a delay, it seems like maybe you need to interact with the editor for the change to take hold, but that's OK.

Thank you!

**Pull request merged**